examples/prom/mixed_nonlinear_diffusion.cpp

```diff
@@ -1602,7 +1615,7 @@ RomOperator::RomOperator(NonlinearDiffusionOperator *fom_,
                          const CAROM::Matrix *Bsinv,
                          const double newton_rel_tol, const double newton_abs_tol, const int newton_iter,
                          const CAROM::Matrix* S_, const CAROM::Matrix *Ssinv_,
-                         const int myid, const bool hyperreduce_source_)
+                         const int myid, const bool hyperreduce_source_, const bool oversampling_)
```

Comment:
It is awkward having to use this variable `oversampling` just to indicate that `Bsinv` should be transposed when multiplying. The alternative of allocating the transpose is inefficient. Maybe a better design would be to add a `transpose` data member in the `Matrix` class, indicating that `transposeMult` should be called for `mult` and `mult` should be called for `transposeMult`. I'm ok with leaving this PR as it is, but if we see a frequent need for this transpose logic, we may want to consider putting the logic in the `Matrix` class.

Reply:
Hi Dylan. Sounds good to me. Let's leave it like this for now and return back to the issue if it becomes a recurrent issue.